### PR TITLE
Issue 740 accessory connection fails while app is in background

### LINKED ISSUE #740: Accessory connection fails when app is backgrounded
### Bug Report
Accessory connection setup fails when app is in background due to the background task being suspended before the app can connect. 

There are 2 issues:
1. When the app is backgrounded, a background task is created so the app can continue to scan for accessories and complete setup once an accessory is detected. This background task is being prematurely ended, freezing any ongoing tasks.
2.  The background task is never restarted if the app re-enters the background.

##### Reproduction Steps
1. Install the SDL_Example app on a phone with iOS 11 installed. Use SDL_iOS branch **release/5.0.0**. Enable iAP connection in the SDL_Example app. 
2. Start the app. Move app to background by pressing the home button
3. Connect app to TDK via USB cord.

##### Expected Behavior
The app icon should show up on the head unit

##### Observed Behavior
The app icon does not show up on the head unit.

##### OS & Version Information
* iOS Version: iOS 11
* SDL iOS Version: SDL_iOS release/5.0.0
* Testing Against: SDL_Example app and a TDK (SYNC 3.0)

### PR DESCRIPTION
**Pull request by NicoleYarroch**
Fixes #740 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Tested with the Streaming Video app installed on iPhone 6 (iOS 10) with a TDK (SYNC 3.0). 
Tested with the Streaming Video app installed on iPhone 6+ (iOS 11) with a TDK (SYNC 3.0). 

### Summary
Fixed accessory setup over a iAP connection failing when app is in background due to the background task being suspended before the setup can complete.
1. Fixed not being able to establish a session with an accessory when the app is backgrounded and when is the app is backgrounded more than once
1. Fixed the app not being able to establish a session with an accessory when the device is first connected to the accessory and the app is closed (but the app has been opened at least once)  

##### Bug Fixes
* Fixed background task ending before accessory setup completed. This was happening in the custom `sessionSetupInProgress` setter. 
* Fixed background task not being started the second time the app is backgrounded. The app now watches for a `UIApplicationDidEnterBackgroundNotification` and starts a background task.
* Fixed errors with the accessory not re-connecting in background because the `retryCounter` already maxed out. 
    * Fixed `retryEstablishSession` method call always failing because  the `retryCounter` was already maxed out.   
    * If there was an error setting up a data stream after closing the control stream, the retry attempt always failed because the `retryCounter` was already maxed out. 
    * `retryCounter` needed to be reset in several places 

##### Tasks to Finish
- [x] iOS 11 issue. Accessory setup will not complete if accessory is connected while the app is closed (but has been opened at least once). Logs show that accessory is connected but the input stream is closed right away as no data is being sent. This seems to work fine with iOS 10. 
- [ ] Test cases are broken in the `SDLStreamingMediaLifecycleManagerSpec`.
 
### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by NicoleYarroch** [comment]
Replaced deprecated methods with new API

**Comment by NicoleYarroch** [comment]
I tested fixes with the Streaming Video Example app on both iOS 10 and iOS 11.

File: SmartDeviceLink/SDLIAPTransport.m [hunk 1]
```diff
@@ -50,19 +51,60 @@ - (instancetype)init {
...
+            [self sdl_backgroundTaskStart];
+        }
+
+        // Get notifications if an accessory connects in future
         [self sdl_startEventListening];
-    }
 
-    SDLLogV(@"SDLIAPTransport Init");
+        // Immediately scan for connected accessories
+        [self sdl_connect:nil];
```
[the remark this answers is not in the document]
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLIAPTransport.m]: Fixed. Removed

**Comment by NicoleYarroch** [comment]
After merging in edits made to branch 5.0.0, there is a new bug in the `SDLStreamingMediaLifecycleManager`. Video streaming works the first time the app is connected to the TDK, but on each subsequent reconnect the streaming life cycle manager never enters the state **Ready**. The accessory connects fine, but it looks like in the `SDLStreamingMediaLifecycleManager` that the  `handleProtocolStartServiceACKMessage` does not get called on reconnects.

**Comment by joeljfischer** [comment]
The `SDLStreamingMediaLifecycleManager`'s protocol property is `nil`.

**Comment by joeljfischer** [comment]
@NicoleYarroch Fixed that issue

**Pull request closed without merging**